### ISSUE TITLE
index sgnature is reported as unused in classes

### ISSUE THREAD

**Issue by armano2** [typescript-eslint/typescript-eslint#262]
This issue was initially reported here: https://github.com/eslint/typescript-eslint-parser/issues/557#issuecomment-449464459

---

This happens with index signatures in class properties as well:

```ts
export default class Foo {
    [key: string]: any;
}
```

```
2:6  error  'key' is not defined  no-undef
```
